index.js

```diff
@@ -218,10 +219,20 @@ module.exports = {
           if (doc.archived && (mode === 'published') && (doc.parkedId !== 'trash')) {
             return false;
           }
+
+          // We don't want the published doc if its draft version is archived
+          if (mode === 'published' && self.archivedDrafts.has(doc.workflowGuid)) {
+            return false;
+          }
+
           doc._id = `${doc.workflowGuid}:${locale}:${mode}`;
           doc.aposDocId = doc.workflowGuid;
           doc.aposLocale = `${locale}:${mode}`;
           doc.aposMode = mode;
+
+          if (doc.archived && mode === 'draft') {
+            self.archivedDrafts.add(doc.workflowGuid);
```

Comment:
@boutell will the published versions always come after the draft one in the loop?
If not, we'll need to do batch deletion of all the published docs that have their draft version archived, at the end of the loop

Reply:
I would suggest doing it that way yes, it is simple. Also though it has to be locale specific.